src/System.Text.Json/src/System/Text/Json/JsonReader.cs

```diff
@@ -107,17 +110,16 @@ public JsonValueType GetJsonValueType()
                 return JsonValueType.Null;
             }
 
-            if (nextByte == '"')
+            if (nextByte == '-' || (nextByte >= '0' && nextByte <= '9'))
             {
-                return JsonValueType.String;
+                return JsonValueType.Number;
             }
 
-            throw new ArgumentOutOfRangeException();
+            throw new FormatException("Invalid json, tried to read char '" + nextByte + "'.");
         }
 
-        public Utf8String GetValue()
+        public Utf8String GetValue(JsonValueType type)
```

Comment:
I agree. The usability seems to over rule the small perf gain. I will revert back to GetValue() and explicitly call GetJsonValueType inside.